### DISCUSSION TITLE
"Get" Component

### DISCUSSION THREAD

**Discussion by T1MM5H** [FakeFishGames/Barotrauma#9550] [Ideas and suggestions]
### Concept: A Wiring Component that can extract Data from Memory/Save and interpret it into a string. 

_Disclaimer: I am aware that there are ways to detect certain pieces of information in the following list by comparing to a preprogrammed memory component. This is clunky and un-intuitive._

**Why it is good for the Game:** 
Would allow for easy Display and Illustration of Data that cannot be recovered from Installations, such as:
-Money in the Bank
-Current Biome
-Crew Size
-PvP Team
-Fatalities
-Round Time
-Round Completions/Campaign Length
-Jovian Radiation %age
-Sub Crush Depth
-Honk Counter
and many more.

If done right, could be vastly expanded upon by adding additional _gets_, potentially even through mods.

**Practical Applications could be Numerous, such as:**
-PvP Round Timers
-Crush Depth Warning Systems
-"X Days Since Last Accident"
-Custom Gamemodes ("Clown Hunt" anyone?)

**Potential Problems:**
-Access violations
-Memory Errors/Leaks
-Security vulnerabilities 
-Privacy Concerns (maybe client-side toggle in network options?)

**Comment by Ek-Videogames** [comment]
+1 [And this same thing from a year ago](https://github.com/Regalis11/Barotrauma/discussions/7595#discussion-3721813)
